kong/router/atc.lua

```diff
@@ -55,32 +52,27 @@ local values_buf = buffer.new(64)
 
 
 local CACHED_SCHEMA
+local HTTP_SCHEMA
+local STREAM_SCHEMA
 do
-  local FIELDS = {
-
-    ["String"] = {"net.protocol", "tls.sni",
-                  "http.method", "http.host",
-                  "http.path",
-                  "http.headers.*",
-                  "http.queries.*",
-                 },
+  local function generate_schema(fields)
+    local s = schema.new()
 
-    ["Int"]    = {"net.port",
-                  "net.src.port", "net.dst.port",
-                 },
-
-    ["IpAddr"] = {"net.src.ip", "net.dst.ip",
-                 },
-  }
-
-  CACHED_SCHEMA = schema.new()
-
-  for typ, fields in pairs(FIELDS) do
-    for _, v in ipairs(fields) do
-      assert(CACHED_SCHEMA:add_field(v, typ))
+    for t, f in pairs(fields) do
+      for _, v in ipairs(f) do
+        assert(s:add_field(v, t))
+      end
     end
+
+    return s
   end
 
+  -- used by validation
+  HTTP_SCHEMA   = generate_schema(fields.HTTP_FIELDS)
+  STREAM_SCHEMA = generate_schema(fields.STREAM_FIELDS)
+
+  -- used by running router
+  CACHED_SCHEMA = is_http and HTTP_SCHEMA or STREAM_SCHEMA
```

Comment:
It seems one of those is not used. Is it?

Reply:
No, HTTP_SCHEMA and STREAM_SCHEMA will be used by validation.